packages/integration-tests/parallel-test.js

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 OmiseGO Pte Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+const RootChain = require('@omisego/omg-js-rootchain')
+const ChildChain = require('@omisego/omg-js-childchain')
+const Web3 = require('web3')
+const fs = require('fs')
+
+const faucet = require('./helpers/faucet')
+const config = require('./test-config')
+
+const MochaParallel = require('mocha-parallel-tests').default
+const mochaParallel = new MochaParallel({
+  enableTimeouts: false,
+  slow: 0,
+  useColors: true,
+  fullStackTrace: true,
+  reporter: 'list'
+})
+
+const allFiles = fs.readdirSync(`${__dirname}/test/`)
+const files = allFiles.filter(i => {
+  // skipped since testing exit queues is impossible with parallel tests calling processExits
+  return i !== 'getExitQueueTest.js'
+})
+
+for (const test of files) {
+  mochaParallel.addFile(`${__dirname}/test/${test}`)
+}
+
+async function setup () {
+  const web3 = new Web3(new Web3.providers.HttpProvider(config.eth_node))
+  const rootChain = new RootChain({ web3, plasmaContractAddress: config.plasmaframework_contract_address })
+  const childChain = new ChildChain({ watcherUrl: config.watcher_url, watcherProxyUrl: config.watcher_proxy_url })
+
+  const start = new Date()
+  for (const filename of files) {
+    await faucet.init(rootChain, childChain, web3, config, filename)
```

Comment:
this can be `Promise.all` and reduce the time significantly `if` we manually increase nonce on the rootchain falcet topup tx. but not required XD

Reply:
Well if the fund account tries to fund 10 faucets in parallel, you couldn’t just increment the nonce since the transactions are not in sequence. You could pre-assign a nonce to each future transaction. However, brings a few problems.. 

1. Depending on the current rootchain/childchain balance of each faucet, you might have to make 0, or up to 3 transactions. You would have to know how many transactions are needed per faucet before assigning nonces.
 2. If a transaction fails, you’ve created a nonce gap  

So.. you would have to know the outcome of which transactions need to be made, and hope they don’t fail to pre-assign nonce values reliably. I guess possible, but ☠️